### DISCUSSION TITLE
Bash: Is it possible to reserve a specific memory area for ImageMagick

### DISCUSSION THREAD

**Discussion by porphyry5** [ImageMagick/ImageMagick#2125] [Help] [answered]
In one of my bash functions I use the following code to send ImageMagick's convert output thru a pipe to tesseract-ocr for conversion to text:

`for a in ocr/*.tif*; do
  echo $a $(date +%T)
  for i in 2 4 8; do
    echo $a >> $i.txt
    convert $a -resize "$i"00% - | tesseract stdin q 2> /dev/null
    cat q.txt >> $i.txt
  done
  mv $a dun/
done
`
which works fine as long as I do nothing else at all while the function is running.

But any other use causes ImageMagick to generate temporary files instead which more than doubles the processing time for each input file. I assume this is because it objects to some new app's use of memory while convert is running. Is there any means in bash by which one can reserve a specific part of memory for one app's exclusive use?

**Comment by snibgo** [comment]
bash is a CLI and scripting language. Your question is really about operating system facilities, and whether you can limit the memory used by other processes so it is available for IM. I suspect the answer is "yes", but I don't know how.

**Comment by porphyry5** [comment]
Thanks for the info.  I've been checking out the linux command list and
came across one called 'nice', which sounds like it will let an app be a
resource-hog, which is essentially what I want to do. So I'll give it a try
in the next few days and report the result.

On Wed, Jun 10, 2020 at 4:51 AM snibgo <notifications@github.com> wrote:

> bash is a scripting language. Your question is really about operating
> system facilities, and whether you can limit the memory used by other
> processes so it is available for IM. I suspect the answer is "yes", but I
> don't know how.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ImageMagick/ImageMagick/discussions/2125#discussioncomment-24456>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACNJ2K4YV5DUN7CWUY5PDD3RV5XVTANCNFSM4NZUK5UA>
> .
>


-- 
Graham Lawrence

**Comment by porphyry5** [comment, marked as answer]
Apparently the command 'nice' is the tool for this problem. I posted the question after noticing that the processing time for this command

`convert $a -resize "$i"00% - | tesseract stdin q 2> /dev/null`

changed from about 1.5 minutes per input file with no other activity on the system, to about 3.3 minutes per input file when I was simultaneously taking multiple screenshots of my web browser.

So I changed the command to

`sudo nice -n -20 convert $a -resize "$i"00% - | tesseract stdin q 2> /dev/null`

for the latest batch and saw my average input file time go from about 1 minute to 1.25 minutes while again doing multiple screenshots of the browser.
